utils/helpers.py

```diff
@@ -104,10 +104,19 @@ def wrapper(*args, **kw):
     return decorator
 
 
-def call_func_by_name(obj, name, *args, **kw):
-    func = getattr(obj, name.lower(), None)
-    if func:
-        return func(*args, **kw)
-    else:
-        raise NameError(
-            'The checkpoint function {} is not defined'.format(name))
+def checkpoint1(check_id):
+    def decorator(func):
+        @functools.wraps(func)
+        def wrapper(*args, **kw):
+            log.info('Start to run test cases:["RHHI-%d"]' % check_id[func.__name__])
+            try:
+                func(*args, **kw)
+            except Exception as e:
+                log.info('func(%s)|| {"RHHI-%d": "failed"}' % (func.__name__, check_id[func.__name__]))
+                log.error(e)
+            else:
+                log.info('func(%s)|| {"RHHI-%d": "passed"}' % (func.__name__, check_id[func.__name__]))
+            finally:
+                log.info('Finished to run test cases:["RHHI-%d"]' % check_id[func.__name__])
+        return wrapper
+    return decorator
```

Comment:
Before we did not consider any other cases id such as "RHHI-137", so we just hard code "RHEVM:" in checkpoint function. The best solution is to change id format as "RHEVM-18317" or "RHHI-137", so you do not need to add another checkpoint1 decorator.
That means:
log.info('func(%s)|| {"%s": "failed"}' % (func.__name__, check_id[func.__name__]))
check_id[func.__name__] will be "RHHI-137" or "RHEVM-18317".
What's your opinion? If yes, could you please help to change the id format in const.py. Thanks.

Reply:
yup, sure. Works for me. Will modify accordingly